stm32-modules/heater-shaker/include/heater-shaker/heater_task.hpp

```diff
@@ -183,46 +224,83 @@ requires MessageQueue<QueueImpl<Message>, Message> class HeaterTask {
             .board_temperature = board.temp_c,
             .pad_a_adc = pad_a.last_adc,
             .pad_b_adc = pad_b.last_adc,
-            .board_adc = board.last_adc};
+            .board_adc = board.last_adc,
+            .power_good = policy.power_good()};
         static_cast<void>(task_registry->comms->get_message_queue().try_send(
             messages::HostCommsMessage(response)));
     }
 
-    auto visit_message(const messages::TemperatureConversionComplete& msg)
+    template <typename Policy>
+    requires HeaterExecutionPolicy<Policy> auto visit_message(
+        const messages::TemperatureConversionComplete& msg, Policy& policy)
         -> void {
+        auto old_error_bitmap = state.error_bitmap;
+        if (!policy.power_good()) {
+            state.error_bitmap |= State::POWER_GOOD_ERROR;
+        }
         handle_temperature_conversion(msg.pad_a, pad_a);
         handle_temperature_conversion(msg.pad_b, pad_b);
         handle_temperature_conversion(msg.board, board);
+        // The error handling wants to accomplish the following:
+        // - Only run if there were any changes in the error state for
+        //   the sensors or the heater pad power driver
+        // - If that change is that the detailed error responses from
+        //   the sensors are now gone, try and reset the power driver
+        // - If that fails, inform upstream
+        // - If the change was that the error latch fired even though it doesn't
+        //   seem like it should have, send that error
+        // - In any case, make sure the overall system state is correct
+
+        auto changes = old_error_bitmap ^ state.error_bitmap;
+        if ((changes & State::PAD_SENSE_ERROR) != 0) {
+            if ((state.error_bitmap & State::PAD_SENSE_ERROR) == 0) {
+                if (policy.try_reset_power_good()) {
+                    state.error_bitmap &= ~State::POWER_GOOD_ERROR;
+                    if (state.error_bitmap == 0) {
+                        state.system_status = State::IDLE;
+                    }
+                } else {
+                    auto error_message =
+                        messages::HostCommsMessage(messages::ErrorMessage{
+                            .code = errors::ErrorCode::
+                                HEATER_HARDWARE_ERROR_LATCH});
+                    static_cast<void>(
+                        task_registry->comms->get_message_queue().try_send(
+                            error_message));
+                    state.system_status = State::ERROR;
+                }
+            } else {
+                state.system_status = State::ERROR;
+            }
+        } else if ((changes & State::POWER_GOOD_ERROR) != 0) {
+            auto error_message =
+                messages::HostCommsMessage(messages::ErrorMessage{
+                    .code = errors::ErrorCode::HEATER_HARDWARE_ERROR_LATCH});
+            static_cast<void>(
```

Comment:
Why is `static_cast<void>` needed?

Reply:
`try_send` returns a `bool` that is `true` if the message was sent, and `false`  otherwise (e.g. because the queue was full, it  has a limited size  since it's statically allocated). That method is annotated `[[nodiscard]]`, which means the compiler will reject calls to  this method that implicitly discard its return value;  you have to explicitly discard it, whichI do here because if this message doesn't get sent it fundamentally doesn't matter since  the state is still set.